### PR TITLE
system/ping[6]: correct the ping return value

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Impact
Return EXIT_FAILURE in the error case

### CODE REVIEW

File: system/ping6/ping6.c [hunk 1]
```diff
@@ -92,8 +101,11 @@ static void show_usage(FAR const char *progname, int exitcode)
 
 static void ping6_result(FAR const struct ping6_result_s *result)
 {
+  FAR struct ping6_priv_s *priv = result->info->priv;
   char strbuffer[INET6_ADDRSTRLEN];
 
+  priv->code = result->code;
```
Comment (yamt) [inline system/ping6/ping6.c:107]: i guess this way many of errors will be overwritten by ICMP_I_FINISH.
is it intended?

Comment (xiaoxiang781216) [reply system/ping6/ping6.c]: Yes, you are right. Please review the updated version again.

File: system/ping/ping.c [hunk 2]
```diff
@@ -224,6 +241,8 @@ int main(int argc, FAR char *argv[])
   info.delay     = ICMP_POLL_DELAY;
   info.timeout   = ICMP_POLL_DELAY;
   info.callback  = ping_result;
+  info.priv      = &priv;
+  priv.code      = ICMP_I_BEGIN;
```
Comment (yamt) [inline system/ping/ping.c:245]: i feel it's cleaner to introduce a dedicated constant than using ICMP_I_BEGIN

**Review by yamt — APPROVED** [review]

File: system/ping/ping.c (hunk 2 above)
Comment (xiaoxiang781216) [reply system/ping/ping.c]: Fix

**Pull request merged**